compiler/rustc_codegen_llvm/src/gotoc/statement.rs

```diff
@@ -313,19 +320,22 @@ impl<'tcx> GotocCtx<'tcx> {
                 // Actually generate the function call, and store the return value, if any.
                 stmts.append(&mut vec![
                     self.codegen_expr_to_place(&p, funce.call(fargs)).with_location(loc.clone()),
-                    Stmt::goto(self.find_label(&target), loc),
+                    Stmt::goto(self.find_label(&target), loc.clone()),
                 ]);
-                Stmt::block(stmts)
+                Stmt::block(stmts, loc)
             }
             ty::FnPtr(_) => {
                 let (p, target) = destination.unwrap();
                 let func_expr = self.codegen_operand(func).dereference();
                 // Actually generate the function call, and store the return value, if any.
-                Stmt::block(vec![
-                    self.codegen_expr_to_place(&p, func_expr.call(fargs))
-                        .with_location(loc.clone()),
-                    Stmt::goto(self.find_label(&target), loc),
-                ])
+                Stmt::block(
+                    vec![
+                        self.codegen_expr_to_place(&p, func_expr.call(fargs))
```

Comment:
This is maybe for another PR, but `codegen_expr_to_place` should really just take a location